src/kdl/serializable.cr

```diff
@@ -0,0 +1,308 @@
+                  default:     ivar.default_value,
+                  nilable:     ivar.type.nilable?,
+                  type:        ivar.type,
+                  converter:   ann[:converter],
+                  presence:    ann[:presence]
+                }
+                all_properties[ivar.id] = arguments_anno
+              %}
+            {% end %}
+          {% elsif ann = ivar.annotation(::KDL::Property) %}
+            {% unless ann[:ignore] || ann[:ignore_deserialize] %}
+              {%
+                property_annos[ivar.id] = {
+                  name:        (ann[:name] || ivar).id.stringify,
+                  has_default: ivar.has_default_value?,
+                  default:     ivar.default_value,
+                  nilable:     ivar.type.nilable?,
+                  type:        ivar.type,
+                  converter:   ann[:converter],
+                  presence:    ann[:presence]
+                }
+                all_properties[ivar.id] = property_annos[ivar.id]
+              %}
+            {% end %}
+          {% elsif ann = ivar.annotation(::KDL::Properties) %}
+            {% unless ann[:ignore] || ann[:ignore_deserialize] %}
+              {%
+                properties_anno = {
+                  id:          ivar.id,
+                  has_default: ivar.has_default_value?,
+                  default:     ivar.default_value,
+                  nilable:     ivar.type.nilable?,
+                  type:        ivar.type,
+                  converter:   ann[:converter],
+                  presence:    ann[:presence]
+                }
+                all_properties[ivar.id] = properties_anno
+              %}
+            {% end %}
+          {% elsif ann = ivar.annotation(::KDL::Child) %}
+            {% unless ann[:ignore] || ann[:ignore_deserialize] %}
+              {%
+                child_annos[ivar.id] = {
+                  name:        (ann[:name] || ivar).id.stringify,
+                  has_default: ivar.has_default_value?,
+                  default:     ivar.default_value,
+                  nilable:     ivar.type.nilable?,
+                  type:        ivar.type,
+                  converter:   ann[:converter],
+                  presence:    ann[:presence],
+                  unwrap:      ann[:unwrap]
+                }
+                all_properties[ivar.id] = child_annos[ivar.id]
+              %}
+            {% end %}
+          {% elsif ann = ivar.annotation(::KDL::Children) %}
+            {% unless ann[:ignore] || ann[:ignore_deserialize] %}
+              {%
+                children_annos[ivar.id] = {
+                  name:        (ann[:name] || ivar).id.stringify,
+                  has_default: ivar.has_default_value?,
+                  default:     ivar.default_value,
+                  nilable:     ivar.type.nilable?,
+                  type:        ivar.type,
+                  converter:   ann[:converter],
+                  presence:    ann[:presence]
+                }
```

Comment:
```cr
unwrap: ann[:unwrap]
```
together with the change further down at `#children` enables child array unwrapping